web/src/app/modules/sugarloaf/components/smart/apply-yaml/apply-yaml.component.ts

```diff
@@ -28,22 +29,23 @@ export class ApplyYAMLComponent implements OnInit {
     },
   };
 
-  constructor() {}
-
-  ngOnInit() {
-    this.isOpen = false;
+  constructor() {
+    ClarityIcons.addIcons(uploadIcon);
   }
 
+  ngOnInit() {}
+
   @HostListener('window:keydown', ['$event'])
   keyEvent(event: KeyboardEvent) {
     if (event.ctrlKey && event.key === 'y') {
       event.preventDefault();
       event.cancelBubble = true;
-      this.isOpen = !this.isOpen;
+      this.toggleModal();
     }
   }
 
   toggleModal() {
-    this.isOpen = true;
+    const yamlModal = document.getElementById('apply-yaml-modal');
+    yamlModal.hidden = !yamlModal.hidden;
   }
 }
```

Comment:
Apply button is now enabled when no text is entered, producing this error when you hit it:
```
ERROR	action/manager.go:193	actionFunc returned err: convert payload to apply yaml request: update is blank	{"component": "action-manager"}
github.com/vmware-tanzu/octant/pkg/action.(*Manager).Dispatch
	/Users/mklanjsek/workspace/octant/pkg/action/manager.go:193
github.com/vmware-tanzu/octant/internal/api.(*WebsocketState).Dispatch
	/Users/mklanjsek/workspace/octant/internal/api/websocket_state.go:187
github.com/vmware-tanzu/octant/internal/api.(*ActionRequestManager).PerformAction
	/Users/mklanjsek/workspace/octant/internal/api/action_request_manager.go:65
github.com/vmware-tanzu/octant/internal/api.(*WebsocketClient).handle.func1
	/Users/mklanjsek/workspace/octant/internal/api/websocket_client.go:191
golang.org/x/sync/errgroup.(*Group).Go.func1
	/Users/mklanjsek/workspace/octant/vendor/golang.org/x/sync/errgroup/errgroup.go:57```

Reply:
Hi @mklanjsek, can you explain me what is this `apply-yaml` and how do we use it? :)

Reply:
@ftovaro  This is the functionality used to directly apply the arbitrary yaml to a namespace and it's activated by hitting the `Apply YAML` button in Octant header. That opens a modal used to enter the yaml content. Unfortunately, after this change,  the `Apply` button inside the modal is enabled when no text is entered, causing the above error. 

Hope that explains it, if not I can record a short video.

Reply:
I'm not seeing this unless we're talking about a different workflow.

![apply-yaml](https://user-images.githubusercontent.com/10288252/115589819-29c41b00-a285-11eb-8e70-43962f0444f1.gif)

Reply:
Yeah, looks like problem is when you hit `Reset`, it does not disable the button, resulting in above error. This is probably existing issue.